test/weights.jl

```diff
@@ -284,81 +325,84 @@ end
-        f([2, 2, 5, 1, 2, 2, 1, 6]),
-        f([0.1, 0.1, 0.8]),
-        f([5, 5, 4, 1]),
-        f([30, 56, 144, 24, 55, 43, 67]),
-        f([0.1, 0.2, 0.3, 0.4, 0.5, 0.6]),
-        f([12]),
-        f([7, 1, 1, 1, 6]),
-        f([1, 0, 0, 0, 2]),
-        f([1, 2, 3, 4, 5]),
-        f([0.1, 0.2, 0.3, 0.2, 0.1]),
-        f([1, 1, 1, 1, 1]),
+        [1, 1/3, 1/3, 1/3, 1],
+        [1, 1, 1, 1, 1],
+        [1, 1/3, 1/3, 1/3, 1, 1],
+        [1/3, 1/3, 1/3, 1, 1, 1],
+        [30, 191, 9, 0],
+        [10, 1, 1, 1, 9],
+        [10, 1, 1, 1, 900],
+        [1, 3, 5, 4, 2],
+        [2, 2, 5, 1, 2, 2, 1, 6],
+        [0.1, 0.1, 0.8],
+        [5, 5, 4, 1],
+        [30, 56, 144, 24, 55, 43, 67],
+        [0.1, 0.2, 0.3, 0.4, 0.5, 0.6],
+        [12],
+        [7, 1, 1, 1, 6],
+        [1, 0, 0, 0, 2],
+        [1, 2, 3, 4, 5],
+        [0.1, 0.2, 0.3, 0.2, 0.1],
+        [1, 1, 1, 1, 1],
     )
     quantile_answers = (
-        [1.0,3.6000000000000005,6.181818181818182,8.2,10.0],
-        [1.0,2.0,4.0,7.0,10.0],
-        [1.0,4.75,8.0,10.833333333333334,15.0],
-        [1.0,4.75,8.0,10.833333333333334,15.0],
-        [0.0,6.1387900355871885,11.600000000000001,15.912500000000001,30.0],
-        [1.0,1.5365853658536586,2.5999999999999996,4.405405405405405,5.0],
-        [1.0,4.239377950569287,4.492918633712858,4.746459316856429,5.0],
-        [30.0,38.75,45.714285714285715,52.85714285714286,60.0],
-        [0.3,0.6903846153846154,1.484,1.7,2.0],
-        [1.0,2.0,2.0,2.0,2.0],
-        [2.8,3.3361111111111112,3.4611111111111112,3.581578947368421,3.7],
-        [45.0,59.88593155893536,100.08846153846153,118.62115384615385,125.0],
-        [2.0,2.0,2.0,2.0,2.0],
-        [2.3,2.3,2.3,2.3,2.3],
-        [-10.0,-5.52,-2.5882352941176467,-0.9411764705882351,2.0],
-        [1.0,1.75,4.25,4.625,5.0],
-        [1.0,1.625,2.3333333333333335,3.25,5.0],
-        [-2.0,-0.5384615384615388,1.5384615384615383,2.6999999999999997,6.0],
-        [-10.0,-10.0,-10.0,1.0,1.0]
+    [1.0, 3.6, 6.18182, 8.2, 10.0],
+    [1.0, 2.0, 4.0, 7.0, 10.0],
+    [1.0, 4.75, 8.0, 10.8333, 15.0],
+    [1.0, 4.75, 8.0, 10.8333, 15.0],
+    [0.0, 4.58167, 9.16335, 14.4976, 20.0],
+    [1.0, 1.53659, 2.6, 4.40541, 5.0],
+    [1.0, 4.23938, 4.49292, 4.74646, 5.0],
+    [30.0, 38.75, 45.7143, 52.8571, 60.0],
+    [0.3, 0.690385, 1.484, 1.7, 2.0],
+    [1.0, 2.0, 2.0, 2.0, 2.0],
+    [2.8, 3.33611, 3.46111, 3.58158, 3.7],
+    [45.0, 59.8859, 100.088, 118.621, 125.0],
+    [2.0, 2.0, 2.0, 2.0, 2.0],
+    [2.3, 2.3, 2.3, 2.3, 2.3],
+    [-10.0, -5.52, -2.58824, -0.941176, 2.0],
+    [1.0, 2.0, 3.0, 4.0, 5.0],
+    [1.0, 1.625, 2.33333, 3.25, 5.0],
+    [-2.0, -0.538462, 1.53846, 2.7, 6.0],
+    [-10.0, -10.0, -10.0, 1.0, 1.0],
     )
     p = [0.0, 0.25, 0.5, 0.75, 1.0]
 
     srand(10)
     for i = 1:length(data)
-        @test quantile(data[i], wt[i], p) ≈ quantile_answers[i]
+        @test quantile(data[i], aweights(wt[i]), p) ≈ quantile_answers[i] atol = 1e-3
```

Comment:
1e-3 sounds quite high, why not keep more precision?

Reply:
This comment hasn't been addressed.

BTW, it would be nice to duplicate each `@test` line to test `pweights` too.